### ISSUE TITLE
Add "Jump to definiton" support

### ISSUE THREAD

**Issue by spacekookie** [ul/kak-lsp#95]
Are there plans to support `lsp-jump-to-definition` so that kakoune tries to open whatever file defines something that the cursor is hovering over (at the line that it's defined at)?

This is something that (at least) the rls supports and would be very useful.

Apologies if there's an issue about this I missed, I didn't find anything.

**Comment by osiox** [comment]
kak-lsp already supports this feature as stated in the readme

> lsp-definition command to go to definition, mapped to gd by default

**Comment by ul** [comment]
Yes, it was my first thought as well that it's already implemented as `lsp-definition`. Let me know if it doesn't work for you or if what you need is command with different behaviour.

**Comment by spacekookie** [comment]
Huh, I didn't realised this was already implemented.

I only tried to run `lsp-definition` on things that weren't defined in my code, i.e. functions and structs that come from crates (or `std`) which could potentially open the appropriate file. Right now it doesn't seem to do that (for me)

**Comment by ul** [comment]
On my system it successfully jumps to std and crates identifiers definitions. I wonder why is it broken on yours. Do you have any errors in Kakoune's `*debug*` buffer? Also, would you mind capturing debug log of kak-lsp? Just do something like `pkill kak-lsp && kak-lsp -s foo -vvv` (where `foo` is Kakoune's session name) and then `gd` in buffer again. Note that `rls` will be killed and restarted by that as well and it takes non-negligible time to index/build project (e.g. around dozen of seconds on MBP 2017 for kak-lsp itself) and unfortunately `rls` just returns empty jump during that time, so you might need to repeat `gd` a bit later for a clean experiment.

**Comment by spacekookie** [comment]
Hey there! Sorry for the delay in reporting this.

So, there's no errors in by `*debug*` buffer. I ran `kak-lsp` in debug mode and captured a whole lot of output. I'm not too intimately familiar with the `lsp`protocol, there was a *lot* of noise in the log (here in it's entirety: [pastebin]). Most of that actually seems to be the rls sending me the appropriate documentation to display (which works really well). But I don't see a `lsp-define` or similar anywhere which makes me believe something else is configured badly?

Maybe this is giving you some ideas what might be wrong.

[pastebin]: https://pastebin.com/rEZpSErs

**Comment by ul** [comment]
Thanks for providing the log! It shows that RLS returns empty response:

```
Oct 08 16:19:32.253 DEBG From server: {"jsonrpc":"2.0","id":54,"result":[]}, module: kak_lsp::language_server_transport:117
```

I wonder if RLS requires some extra components to give proper jumps. But I'm pretty sure you installed yours with the same command `rustup component add rls-preview rust-analysis rust-src` as me. (But if not it worth trying to re-install RLS this way)

Could you try to jump into std/crate definition in VS Code using [this  extension](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust)? I wonder if it works there.

**Comment by ul** [comment]
@spacekookie any updates?

**Comment by spacekookie** [comment]
Hey there, sorry for the delay. I'll try out the `vscode` plugin. I haven't been around code editors a lot recently :grimacing:

**Comment by ul** [comment]
Feel free to re-open issue once you get back to code editor 😉

**Comment by spacekookie** [comment]
Hey, I just tried this and it also doesn't work in `vscode`. I'm not really sure where to find the debug output of the rls extention in vscode

**Comment by ul** [comment]
Thanks for update! Then it looks like an RLS issue in your environment, might worth posting ticket on https://github.com/rust-lang-nursery/rls or asking Rust subreddit.
